React/03_class/shopping_cart/src/components/Products/reducer.js

```diff
@@ -27,7 +30,39 @@ export const product = (state = initialState, action) => {
 
     case "SHOULD_SHOW_FILTERED_PRODUCTS":
       return {...state, shouldShowFilteredProducts: action.payload};
-    
+
+    case "SORT_BY_INCREASING_ORDER":
+      if(state.shouldShowFilteredProducts){
+        state.filteredProducts = state.filteredProducts.sort((a, b) => parseFloat(a.price) - parseFloat(b.price));
+        return {...state, filteredProducts: state.filteredProducts };
+      }
+      state.products = state.products.sort((a, b) => parseFloat(a.price) - parseFloat(b.price));
+      return {...state, products: state.products };
+
+    case "SORT_BY_DECREASING_ORDER":
+      // if we search, then we have to show filtered products
+      if(state.shouldShowFilteredProducts){
+        state.filteredProducts = state.filteredProducts.sort((a, b) => parseFloat(b.price) - parseFloat(a.price));
+        return {...state, filteredProducts: state.filteredProducts };
+      }
+      state.products = state.products.sort((a, b) => parseFloat(b.price) - parseFloat(a.price));
+      return {...state, products: state.products };
+
+    case "SORT_BY_RATING_HIGH":
+      if(state.shouldShowFilteredProducts){
+        state.filteredProducts = state.filteredProducts.sort((a, b) => parseFloat(a.rating.rate) - parseFloat(b.rating.rate));
```

Comment:
Sort function could have been a single function which accepts two params: attribute (rating.rate) and order (up or down). Instead we're maintaining four different functions. What happens when we want to sort by more columns or use it for a different kind of resource?